### PR TITLE
[PRC] Warning elimination and some minor reformating

### PR DESCRIPTION
**Pull request by JJ**
I'm also working on new tests, but some stuff needs to be solved first.

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -1,17 +1,21 @@
 Term::EditLine
 ==============
 
-Term::EditLine is a object oriented perl interface to the NetBSD editline
-library. To use the module you will have to install this library. Read
+Term::EditLine is an object oriented perl interface to the NetBSD editline
+library. To use the module you will have to install this library first. Read
```
Comment (plicease) [inline README]: This copy isn't accurate anymore, if the `libedit` isn't available then `Alien::Editline` will be used to download and build the library for you.

File: example/synopsis.pl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Term::EditLine qw(CC_EOF);
+
+use v5.14;
```
Comment (plicease) [inline example/synopsis.pl]: I am not sure why 5.14 is necessary?

File: README (hunk 1 above)
Comment (JJ) [reply README]: I just added "first". Do you mean that sentence should be eliminated altogether?

File: example/synopsis.pl (hunk 2 above)
Comment (JJ) [reply example/synopsis.pl]: Well, it's modern Perl, but in this case it was just to use "say". Can be eliminated.

Comment (plicease) [reply example/synopsis.pl]: To use say we'd only need `use v5.10`.

nit: I'd prefer any `use <perl version>` statement to go immediately after the `use warnings`.

```perl
use strict;
use warnings;
use 5.010;
```

File: example/synopsis.pl [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+my $el = Term::EditLine->new('progname');
+$el->set_prompt ('# ');
+
+$el->add_fun ('bye','desc',sub { say "\nbye"; return CC_EOF; });
+
+$el->parse('bind','-e');
+$el->parse('bind','^D','bye');
+
+while (defined($_ = $el->gets())) {
```
Comment (plicease) [inline example/synopsis.pl]: nit: I know that this is from the original POD (which I didn't write), but while we are in here, can we change this here (and in the POD) to:

```perl
while (defined( my $line = $el->gets )) {
...
```

this code has unnecessary parans and I am not a fan of using the topic variable like this in blocks of more than one line.

File: README (hunk 1 above)
Comment (plicease) [reply README]: I'd like to reword it to represent how it works now.  Something like "This module will fallback on Alien::Editline if your system does not have libedit already installed".

File: README [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
 the pod documentation for more information on this module.
 
 INSTALLATION
 
+Install libedit, for instance with this command in Ubuntu:
```
Comment (plicease) [inline README]: nit: prefer to refer to Debian here rather than Ubuntu as the latter is derived from the former.  please reword to something like "in Debian" or "in Debian based distributions (including Ubuntu)".

Please also add the instructions for RedHat based distros:

```
sudo yum install libedit-devel
```

**Comment by plicease** [comment]
I like this overall, I just had a few nits.

It would be good to use the [InsertExample] dzil plugin 

https://metacpan.org/pod/Dist::Zilla::Plugin::InsertExample

For the synopsis example so that it only needs to be changed in one place.

**Comment by plicease** [comment]
Great!  Thanks.
